rubrik_cdm/cluster.py

```diff
@@ -606,3 +608,58 @@ def read_only_authorization(self, username, timeout=15):
 
         self.log("read_only_authorization: Granting read-only privilages to user '{}'.".format(username))
         return self.post("internal", "/authorization/role/read_only_admin", config, timeout)
+
+    def cluster_node_id(self):
+        """ Returns a list of node ids from all the nodes in the cluster.
+
+                Arguments:
+                    None
+
+                Keyword Arguments:
+                    None
+
+                Returns:
+                    dict -- The full API response from `POST /internal/authorization/role/read_only_admin`.
+                """
+        self.log("cluster_node_id - Getting all the node ids from the from the cluster")
+        node_id = self.get('internal', '/node')
+        return node_id
+
+    def cluster_support_tunnel(self,action):
+        """ Function that can check the status, enable or disable the support tunnel.
+
+                        Arguments:
+                            Action - with state of (Status,Enable,Disable)
+
+                        Keyword Arguments:
+                            Action with state of "Status" returns the current status of the support tunnel
+                            Action with state of "Enable" checks the status first then enables the support tunnel if needed
+                            Action with state of "Disable" will disable the current tunnel
+
+                        Returns:
+                            dict -- The full API response from `POST /internal/node/me/support_tunnel`.
+                        """
+        if action =="Status":
+            self.log("cluster_support_tunnel - Get Status of Cluster Support Tunnel")
+            tunnel = self.get('internal', '/node/me/support_tunnel')
+            return tunnel
+
+        elif action == "Enable":
+            self.log("cluster_support_tunnel - Check Status of Cluster Support Tunnel and Enable it if needed")
+            check_tunnel = self.cluster_support_tunnel("Status")
+            if check_tunnel['isTunnelEnabled'] == False:
+                config = {}
+                config['isTunnelEnabled'] = True
+                config['inactivityTimeoutInSeconds'] = 0
+                tunnel = self.patch('internal', '/node/me/support_tunnel', config)
+                return tunnel
+
+            else:
+                return("Tunnel is already enabled")
```

Comment:
`return("Tunnel is already enabled")` for the impotence check we also want to include "No change required." at the start of the message. This makes sure there is a uniform way of checking for them

Reply:
Still need to update the message here